lib/blockchain.go

```diff
@@ -303,6 +323,32 @@ func NewBlockNode(
 		CumWork:          cumWork,
 		Header:           header,
 		Status:           status,
+		// All blocks have a committed status in PoW.
+		CommittedStatus: COMMITTED,
+	}
+}
+
+// NewPoSBlockNode is a new helper function to create a block node
+// as we need to control the value of the CommittedStatus field.
+func NewPoSBlockNode(
+	parent *BlockNode,
+	hash *BlockHash,
+	height uint32,
+	difficultyTarget *BlockHash,
+	cumWork *big.Int,
+	header *MsgDeSoHeader,
+	status BlockStatus,
+	committedStatus CommittedBlockStatus) *BlockNode {
+
+	return &BlockNode{
+		Parent:           parent,
+		Hash:             hash,
+		Height:           height,
+		DifficultyTarget: difficultyTarget,
+		CumWork:          cumWork,
```

Comment:
What's the safes default value we can set for these two fields? Seems like `nil` should be fine since they're not used in ProcessBlockPoS
```suggestion
		DifficultyTarget: nil,
		CumWork:          nil,
```